velox/connectors/hive/HiveDataSink.h

```diff
@@ -238,6 +240,20 @@ class HiveInsertTableHandle : public ConnectorInsertTableHandle {
             "ensureFiles is not supported with partition keys in the data");
       }
     }
+
+    if (partitionKeyOrder_.size() > 0) {
+      folly::F14FastSet<std::string> partitionKeyNames(
+          partitionKeyOrder_.begin(), partitionKeyOrder_.end());
+      for (const auto& inputColumn : inputColumns_) {
+        if (inputColumn->isPartitionKey()) {
+          VELOX_CHECK(partitionKeyNames.erase(inputColumn->name()) == 1);
+        }
+      }
+
+      VELOX_CHECK(
+          partitionKeyNames.empty(),
+          "Ensure the partitionKeyOrder contains all the partition keys in inputColumns_");
```

Comment:
`partitionKeyOrder does not contain all the partition keys` should be sufficient.